doc: ug_edge_impulse: Add downloading public project instruction

alstrzebonski opened the pull request:
Adds instruction on how to download public Edge Impulse project.
Fixes some inconsistencies in instruction on how to download private
Edge Impulse project.

Jira: NCSDK-12104

Signed-off-by: Aleksander Strzebonski <aleksander.strzebonski@nordicsemi.no>

MarekPieta commented on doc/nrf/ug_edge_impulse.rst:
```diff
@@ -109,11 +110,33 @@ Complete the following steps to do this:
+   .. parsed-literal::
+      :class: highlight
+
+      CONFIG_EDGE_IMPULSE_URI="https:\ //studio.edgeimpulse.com/v1/api/*XXXXX*/deployment/download?type=zip&modelType=int8"
```
I would add a note explaining why modelType=int8 is added here (that we can only download a quantized EON compiled model).

MarekPieta reviewed (approved):
2 small fixes

greg-fer commented:
I'm busy with the 1.9.0 release tasks and the deadline is upon us, so I'll have a look at this PR early next week for after the 1.8.0 release.

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-6368
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

greg-fer commented on doc/nrf/ug_edge_impulse.rst:
```diff
@@ -109,11 +115,44 @@ Complete the following steps to do this:
+      https:\ //studio.edgeimpulse.com/v1/api/*XYZ*/versions/public
+
+   Set *XYZ* to the ID shown in the Edge Impulse Studio in the :guilabel:`Project info` panel under :guilabel:`Dashboard`.
+   The ID of the public project is displayed next to the "publicProjectId" field.
+
+   .. figure:: images/ei_project_id.png
+      :scale: 50 %
+      :alt: Project ID in Edge Impulse studio dashboard
+
+      Project ID in Edge Impulse studio dashboard
```
Duplicated the figure from line 112, but we need an updated one, so that the publicProjectId is visible.

greg-fer commented:
Please also include a changelog entry.

greg-fer reviewed (changes requested):
Please check and squash my review. Pay attention to the project ID -- can we include this value in public docs?

alstrzebonski commented:
> Please check and squash my review. Pay attention to the project ID -- can we include this value in public docs?

I don't see a problem with the ID being shown. It's just the ID to my public EI project cloned from NCS EI project.
I squashed the review and changed just one comment at the end:
> This URI includes the ``modelType=int8`` parameter because from public Edge Impulse projects you can only download quantized models created with Edge Impulse's EON Compiler.
I think it is worth mentioning that this information applies only to public EI projects. Models from private projects don't have to neccessary be quantized or use EON Compiler.
